File: php/src/BinarySearchTree/Node.php
<?php

namespace Claudiosilvajunior\Php\BinarySearchTree;

class Node
{
    public function __construct(
        private int $label,
        private ?self $left = null,
        private ?self $right = null,
    ) {
    }

    public function getLabel(): int
    {
        return $this->label;
    }

    public function setLabel(int $label): void
    {
        $this->label = $label;
    }

    public function getLeft(): ?Node
    {
        return $this->left;
    }

    public function setLeft(?Node $left): void
    {
        $this->left = $left;
    }

    public function getRight(): ?Node
    {
        return $this->right;
    }

    public function setRight(?Node $right): void
    {
        $this->right = $right;
    }
}

File: java/src/main/java/org/example/binarySearchTree/Node.java
package org.example.binarySearchTree;

public class Node {
    private int label;
    private Node left = null;
    private Node right = null;

    public Node(int label) {
        this.label = label;
    }

    public Node(int label, Node left, Node right) {
        this.label = label;
        this.left = left;
        this.right = right;
    }

    public int getLabel() {
        return label;
    }

    public void setLabel(int label) {
        this.label = label;
    }

    public Node getLeft() {
        return left;
    }

    public void setLeft(Node left) {
        this.left = left;
    }

    public Node getRight() {
        return right;
    }

    public void setRight(Node right) {
        this.right = right;
    }
}
